server/app/routes/utils.js

```diff
@@ -0,0 +1,196 @@
+  createAdminUser,
+  createGuest,
+  createLocalUser,
+  Credentials,
+  db,
+  describeIt,
+  ensureAuthenticated,
+  _err,
+  forceSyncDb,
+  getApp,
+  getModel,
+  getModels,
+  joinRootPath,
+  logInAgent,
+  logOutAgent,
+  respondWith404,
+  responder,
+  routerUse,
+  Sequelize,
+  should,
+  supertest,
+  syncDb
+}
+
+
+
+// dependencies
+const path = require( 'path' );
+const bluebird = bluebirdForTests();
+
+//declarations
+const appPath = joinRootPath( "./server/app" );
+const dbPath = joinRootPath( "./server/db" );
+global.paths = {routerUtils: joinRootPath( './server/app/routes/utils.js')}
+
+function addUser( credentials ) {
+
+  // console.log(credentials);
+  return getModel('user').create( credentials )
+  .then( result => {
+    
+  })
+  .catch( err => console.log('oh no there was an error here!', err))
+}
+
+function bluebirdForTests() {
+
+  const Bluebird = require( 'bluebird' );
+  return Bluebird;
+}
+
+function createAdminUser( userInfo ) {
+  userInfo.isAdmin = true;
+  return createLocalUser( userInfo )
+}
+
+function createAgent( ) {
+
+  return supertest().agent( getApp() );
+}
+
+function createGuest() {
+  return createAgent();
+}
+
+function createLocalUser( userInfo ) {
+
+  return addUser(userInfo).then( ()=> {
+
+    return createAgent()})
+}
+
+function Credentials( username, password, email ) {
+  return {
+    username,
+    password,
+    email
+  }
+}
+
+function db() {
+
+  return require( dbPath )
+}
+
+function describeIt( describeStmt, itStmt, callback = function(){}, callWithDone ){
+  return describe( describeStmt, function() {
+    it( itStmt, callback( callWithDone ? done : undefined ) )
+  })
+}
+
+function ensureAuthenticated(req, res, next) {
+    if (req.isAuthenticated()) {
+        next();
+    } else {
+        res.status(401).end();
+    }
+}
+
+function _err( err ) {
+  return {
+    err
+  };
+}
+
+function forceSyncDb(){
+  return syncDb(true);
+}
+
+function getApp() {
+
+  let theDb = db();
+  let app = require( appPath )(theDb)
+  return app;
+}
+
+function getUserModel(){
+  return require('./../../db/models/user')
+}
+
+function getModel( model ) {
+
+  const _db = db();
+  // console.dir(_db.models.user);
+  return _db.models[model];
+}
+
+function getModels( models ) {
+  return models.map( model => getModel( model ) )
+}
+
+function joinRootPath( targetPath ) {
+  return path.join( './../../../', targetPath )
+}
+
+function logInAgent( agent, userInfo ) {
+  return agent
+    .post( '/login' )
+    .send( userInfo );
+}
+
+function logOutAgent( agent ) {
+  return agent.post( '/logout' )
+}
+function respondWith404(router){
+  router.use(function (req, res) {
+      res.status(404).end();
+  });
+}
+
+class responder {
+  constructor(router){
+    this.router = router;
+  }
+  with404(){
+    respondWith404(this.router);
+  }
+}
+
+function routerUse(router){
+  return function use(route, path){
+    return router.use( route, require(path) )
+  }
+}
+
+function Sequelize() {
```

Comment:
right. it doesn't really save time, but I figured let's just have one utils file that has commonly used dependencies in it.